manifests/base.pp

```diff
@@ -110,9 +126,11 @@
     exec{'shorewall6_check':
       command     => 'shorewall6 check',
       refreshonly => true,
+      path        => $path,
       require     => Package['shorewall6'],
     } ~> exec{'shorewall6_try':
-      command     => 'shorewall6 try /etc/shorewall6/puppet',
+      command     => 'shorewall6 try /etc/shorewall6',
```

Comment:
why? you will not test anymore the right thing

Reply:
shorewall try search shorewall.conf. In manifest base you declare:     
file { '/etc/shorewall6/shorewall6.conf':

shorewall6 try /etc/shorewall6/puppet
return error:
   ERROR: /etc/shorewall6/puppet/shorewall6.conf does not exist!
shorewall6 try /etc/shorewall6 work fine for me.

Reply:
Which version is this?

I get:

```
# shorewall6 try /etc/shorewall6/
Compiling using Shorewall 5.1.10.2...
Processing /etc/shorewall6/params ...
Processing /etc/shorewall6/shorewall6.conf...
Loading Modules...
Compiling /etc/shorewall6/zones...
   ERROR: No IP zones defined /etc/shorewall6/zones (EOF)
```
vs.
```
# shorewall6 try /etc/shorewall6/puppet/
Compiling using Shorewall 5.1.10.2...
Processing /etc/shorewall6/puppet/params ...
Processing /etc/shorewall6/shorewall6.conf...
Loading Modules...
Compiling /etc/shorewall6/puppet/zones...
Compiling /etc/shorewall6/puppet/interfaces...
Determining Hosts in Zones...
Locating Action Files...
Compiling /etc/shorewall6/puppet/policy...
Adding Anti-smurf Rules
Compiling TCP Flags filtering...
Compiling MAC Filtration -- Phase 1...
Compiling /etc/shorewall6/puppet/rules...
Compiling /usr/share/shorewall/action.dropNotSyn for chain dropNotSyn...
Compiling /etc/shorewall6/conntrack...
Compiling MAC Filtration -- Phase 2...
Applying Policies...
Compiling /usr/share/shorewall/action.AllowICMPs for chain AllowICMPs...
Compiling /usr/share/shorewall/action.Broadcast for chain Broadcast...
Compiling /usr/share/shorewall/action.Multicast for chain Multicast...
Generating Rule Matrix...
Optimizing Ruleset...
Creating ip6tables-restore input...
Shorewall configuration compiled to /var/lib/shorewall6/.reload
   Currently-running Configuration Saved to /var/lib/shorewall6/.try
Reloading...
Reloading Shorewall6....
Initializing...
Processing /etc/shorewall6/init ...
Setting up Proxy NDP...
Preparing ip6tables-restore input...
Running /sbin/ip6tables-restore --wait 60...
Processing /etc/shorewall6/start ...
Processing /etc/shorewall6/started ...
done.
```

Reply:
Oh..
My solution don’t work on major version 4. I’m fixed this in new commit, thanks.